sdk/servicebus/service-bus/test/backupMessageSettlement.spec.ts

```diff
@@ -0,0 +1,561 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+import {
+  ServiceBusMessage,
+  QueueClient,
+  SubscriptionClient,
+  ServiceBusClient,
+  TopicClient,
+  SessionReceiver,
+  Receiver
+} from "../src";
+import {
+  TestClientType,
+  TestMessage,
+  purge,
+  getServiceBusClient,
+  getSenderReceiverClients
+} from "./utils/testUtils";
+import { Sender } from "../src/sender";
+import { DispositionType, ReceiveMode, SendableMessageInfo } from "../src/serviceBusMessage";
+
+const should = chai.should();
+chai.use(chaiAsPromised);
+
+describe("Backup message settlement - Through ManagementLink", () => {
+  let sbClient: ServiceBusClient;
+
+  let senderClient: QueueClient | TopicClient;
+  let receiverClient: QueueClient | SubscriptionClient;
+  let deadLetterClient: QueueClient | SubscriptionClient;
+  let sender: Sender;
+  let receiver: Receiver | SessionReceiver;
+
+  async function testPeekMsgsLength(
+    client: QueueClient | SubscriptionClient,
+    expectedPeekLength: number
+  ): Promise<void> {
+    const peekedMsgs = await client.peek(expectedPeekLength + 1);
+    should.equal(
+      peekedMsgs.length,
+      expectedPeekLength,
+      "Unexpected number of msgs found when peeking"
+    );
+  }
+
+  async function beforeEachTest(
+    senderType: TestClientType,
+    receiverType: TestClientType
+  ): Promise<void> {
+    const useSessions = receiverType > 5 && receiverType < 12 ? true : false;
```

Comment:
Well this check scares me, and actually looks like a case where a bit flag could be useful. Since that might mean making a lot of changes to this PR, could you add a comment to `TestClientType` that states that tests depend on this enum not changing?

Reply:
I will add a comment.